core/src/actors/peers_manager.rs

```diff
@@ -20,6 +25,37 @@ pub struct PeersManager {
     peers: Peers,
 }
 
+impl PeersManager {
+    /// Method to periodically dump peers into storage
+    fn storage_peers(&self, ctx: &mut Context<Self>, storage_peers_period: Duration) {
+        // Schedule the discovery_peers with a given period
+        ctx.run_later(storage_peers_period, move |act, ctx| {
+            // Get storage manager address
+            let storage_manager_addr = System::current().registry().get::<StorageManager>();
```

Comment:
Can't see why would it be necessary to get this address every time. Why not querying it and storing in the actor state upon start up?

Reply:
Are we sure that the address will not change when the storage actor restarts? I think that the added code complexity does not justify the potential performance improvement because:

* In actix, the address is stored in a hashmap

* This function is called once every 30 seconds

And if we decide to store the address in the actor state, should all the actors store addresses to their most used actors?

Reply:
> And if we decide to store the address in the actor state, should all the actors store addresses to their most used actors?

That depends on whether we assume manager actors' lifespan to be everlasting or that some of them can die while others are still alive and thus need fresh addresses.

None of those assumptions are explicitly stated anywhere. If we all agree that we are better off taking the second assumption, I withdraw my former comment.

Reply:
If you don't mind, I think we could discuss it tomorrow. As @tmpolaczyk wrote, if we save the addresses in the state, we should also modify other actors to do the same. What do you think we create an issue for that and we finish this PR?